src/LevelCreator.cpp

```diff
@@ -0,0 +1,114 @@
+#include "LevelCreator.h"
+
+
+/* 
+Level Creator class serves as an engine for reading in a specific level text file
+and loading the proper tiles for that level to display to the player.
+*/
+
+LevelCreator::LevelCreator() {
+}
+
+void LevelCreator::init(){
+
+}
+
+void LevelCreator::LoadLevel(int level){
+    //This line will change based on "level", for now just using MapTest
+    // std::ifstream openfile("../res/MapTest.txt");
+
+    //This line will change based on "level", for now just using Sample Level XML
+    tinyxml2::XMLDocument map;
+    tinyxml2::XMLError eResult = map.LoadFile("../res/practice_level.xml");
+    if (eResult != tinyxml2::XML_SUCCESS) {
+        std::cout << "error loading file" << std::endl;
+    };
+
+    tinyxml2::XMLHandle docHandle(&map);
+
+    tinyxml2::XMLElement *entry = docHandle.FirstChildElement("Actors").ToElement();
+
+
+    if (entry) {
+        for (tinyxml2::XMLNode *node = entry->FirstChildElement(); node; node = node->NextSibling()){
+
+            tinyxml2::XMLElement *e = node->ToElement();
+
+            std::cout << e->Value() << std::endl;
+
+            if (std::strncmp(e->Value(), "Platform", 9) == 0) {
```

Comment:
If you are doing a simple string compare, just use "==" or at the very least I believe you can use strcmp since you are comparing the whole word and not a specific amount of characters

Reply:
I believe in C and C++ you cannot use the "==" for string comparisons. I can change it to strcmp though

Reply:
In Java you can't use "==" since that operator compares the memory address instead of the string's contents, but in C++ "==" essentially just calls the compare function and checks that it equals 0. So "==" in C++ does compare the string's contents, so you should use "==" since it's more readable

Reply:
That doesn't work here though because the Value() method for tinyxml returns a char* that represents an array of chars and c++ treats literal strings as char arrays as well. Thus, it does an == between two char arrays and the == method for string never gets called because neither of them are actually strings. std::strcmp accepts strings and char arrays for input, which is why it works.